src/Elements/types.jl

```diff
@@ -0,0 +1,180 @@
+const HEADERS   = ["#", "##", "###", "####", "#####", "######"]
+const EMPHASIS  = ["*", "**"]
+
+const CATEGORIES = [:module,    :function, :method, :macro, :type,
+                    :typealias, :global,   :symbol, :tuple, :aside]
+
+
+## Config Nodes
+abstract ConfigNode
+
+type Config <: ConfigNode
+    objfilter               :: Any
+    style_obj               :: ASCIIString
+
+    const defaults = Dict{Symbol, Any}(
+        :objfilter                  => CATEGORIES,
+        :style_obj                  => "####",
```

Comment:
I've been thinking we should have something a bit more versatile that just a string.

Passing in an anonymous function as the styling would be neat.

```jl
page(
    Docile,
    Docile.Formats,
    style_obj = s -> string("#### ", s),
)
```

Note we should be able to pass arbitrary keyword arguments to a node to add ``config`` data. Avoids needing to explicitly write `config(...)` every time.